include/userobjects/VelocityUpdaterBase.h

```diff
@@ -0,0 +1,46 @@
+//* This file is part of FENIX: Fusion Energy Integrated Multiphys-X,
+//* A multiphysics application for modeling plasma facing components
+//* https://github.com/idaholab/fenix
+//*
+//* FENIX is powered by the MOOSE Framework
+//* https://www.mooseframework.inl.gov
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// MOOSE Includes
+#include "Ray.h"
+#include "GeneralUserObject.h"
+#include "RayTracingStudy.h"
+
+class VelocityUpdaterBase : public GeneralUserObject
+{
+public:
+  VelocityUpdaterBase(const InputParameters & parameters);
+
+  static InputParameters validParams();
+  // setting all of these methods to do nothing since this class will only
+  // provide a method to be called by a ray tracing study
+  virtual void initialSetup() override {}
+  virtual void residualSetup() override {}
+  virtual void jacobianSetup() override {}
+  virtual void meshChanged() override {}
+  virtual void timestepSetup() override {}
+  virtual void initialize() override {}
+  virtual void finalize() override {}
+  virtual void execute() override {}
```

Comment:
```suggestion
  /**
   * Unused methods
   */
  ///@{
  virtual void initialize() override {}
  virtual void finalize() override {}
  virtual void execute() override {}
  ///@}
```

I think that only the final three need to be overriden. The first ones have default, empty implementations. Also fixes the doxygen comment.

Reply:
This change was also made